internal/armada/repository/job.go

```diff
@@ -838,36 +839,60 @@ func (repo *RedisJobRepository) GetQueueJobIds(queueName string) ([]string, erro
 }
 
 func (repo *RedisJobRepository) GetActiveJobIds(queue string, jobSetId string) ([]string, error) {
+	return repo.GetJobSetJobIds(queue, jobSetId, &JobSetFilter{
+		IncludeLeased: true,
+		IncludeQueued: true,
+	})
+}
+
+type JobSetFilter struct {
+	IncludeQueued bool
+	IncludeLeased bool
+}
+
+func (repo *RedisJobRepository) GetJobSetJobIds(queue string, jobSetId string, filter *JobSetFilter) ([]string, error) {
+	var queuedIdsCommandResult *redis.StringSliceCmd
+	var leasedIdsCommandResult *redis.StringSliceCmd
 
 	tx := repo.db.TxPipeline()
```

Comment:
Not part of this review specifically- but are you sure this TX is doing anything.  My understanding of Redis Transactions was that you had to pass in a list of keys to `Watch` before you call `Exec` if you want transactional behaviour.

Reply:
I think that it is doing is trying to get a consistent snapshot of the ids.

A transaction in theory says all these commands will run without anyone running in between them - so that state change change between them.

I believe this is how it work - without having to do watches. However I could be wrong, but that is what it is trying to achieve.

Reply:
no actually you are right- you only need a watch for other keys